tests/zfs-tests/tests/functional/mmp/mmp_on_zdb.ksh

```diff
@@ -0,0 +1,81 @@
+#!/bin/ksh
+
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2018 Lawrence Livermore National Security, LLC.
+# Copyright (c) 2018 by Nutanix. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/mmp/mmp.cfg
+. $STF_SUITE/tests/functional/mmp/mmp.kshlib
+
+#
+# Description:
+# zdb will work while multihost is enabled.
+#
+# Strategy:
+# 1. Create a pool
+# 2. Enable multihost
+# 3. Run zdb -d with pool and dataset arguments.
+# 4. Create a checkpoint
+# 5. Run zdb -kd with pool and dataset arguments.
+# 6. Discard the checkpoint
+# 7. Export the pool
+# 8. Run zdb -ed with pool and dataset arguments.
+#
+
+function cleanup
+{
+	datasetexists $TESTPOOL && destroy_pool $TESTPOOL
+	for DISK in $DISKS; do
+		zpool labelclear -f $DEV_RDSKDIR/$DISK
+	done
+	log_must mmp_clear_hostid
+}
+
+log_assert "Verify zdb -d works while multihost is enabled"
+log_onexit cleanup
+
+verify_runnable "global"
+verify_disk_count "$DISKS" 2
+
+default_mirror_setup_noexit $DISKS
+log_must mmp_set_hostid $HOSTID1
+log_must zpool set multihost=on $TESTPOOL
+log_must zfs snap $TESTPOOL/$TESTFS@snap
+
+log_must zdb -d $TESTPOOL
+log_must zdb -d $TESTPOOL/
+log_must zdb -d $TESTPOOL/$TESTFS
+log_must zdb -d $TESTPOOL/$TESTFS@snap
+
+log_must zpool checkpoint $TESTPOOL
```

Comment:
0.7 doesn't have `checkpoint` functionality, so test fails.

Reply:
You should be able to just remove this block from tests/zfs-tests/tests/functional/mmp/mmp_on_zdb.ksh 

```
log_must zpool checkpoint $TESTPOOL
log_must zdb -kd $TESTPOOL
log_must zdb -kd $TESTPOOL/
log_must zdb -kd $TESTPOOL/$TESTFS
log_must zdb -kd $TESTPOOL/$TESTFS@snap
log_must zpool checkpoint -d $TESTPOOL
```

Reply:
That block is there purely to test that zdb does the right thing when checkpoints are present.  Since they cannot be present in 0.7, because the functionality does not exist there, that test scenario has no purpose.  

Also, removal of that block should not affect what comes after it.

Reply:
it is trivial to check if a pool property or feature exists and do the right thing, see function get_pool_prop

Reply:
> it is trivial to check if a pool property or feature exists and do the right thing, see function get_pool_prop

It's good to know about get_pool_prop.  In this case, since it's a back port to a branch that we know will never have checkpoints, my thinking was that adding the conditional logic to the test would be misleading, in that it would imply that checkpoints might exist.  Do you think that's the wrong choice?

Reply:
you can also have the test respond with skip, instead of pass/fail

Reply:
I don't really have an opinion either way, so I'll defer to @ofaaland and delete the section of code, since it's his commit.